text/0001-rfc-process.md

```diff
@@ -1,98 +1,125 @@
+* The relevant sub-team discuss the RFC pull request, as much as possible in the comment thread of the pull request itself. Offline discussion will be summarized on the pull request comment thread.
+* RFCs rarely go through this process unchanged, especially as alternatives and drawbacks are shown. As an author uou can make edits, big and small, to the RFC to clarify or change the design,but make changes as new commits to the pull request, and leave a comment on the pull request explaining your changes. Specifically, do not squash or rebase commits after they are visible on the pull request.
 
-At this point, whoever merges the PR should:
-* Assign an id, using the PR number of the RFC pull request. (If the RFC has multiple pull requests associated with it, choose one PR number,
-  preferably the minimal one.)
-* Rename the file to `grin-rfc-xxxx.md` and merge into `/text`. (where
-xxxx is the RFC ID)
-* Fill in the remaining metadata in the RFC header.
-* Add an entry in the RFC List in `README.md`, with a status of Draft.
+#### Final Comment Period (FCP)
 
-When the RFC is in 'Draft' stage, it is important to refine details, build consensus among the community for the RFC and integrate useful
-feedback. RFCs that have broad support are much more likely to make progress than those that don't receive any comments.
+* At some point, a member of the sub-team will propose a "motion for final comment period" (FCP), along with a *disposition* for the RFC (merge, close, or postpone).
+* This step is taken when enough of the tradeoffs have been discussed that the sub-team is in a position to make a decision. That does not require consensus amongst all participants in the RFC thread (which is usually impossible). However, the argument supporting the disposition on the RFC needs to have already been clearly articulated, and there should not be a strong consensus *against* that position outside of the sub-team. Sub-team members use their best judgment in taking this step, and the FCP itself ensures there is ample time and notification for stakeholders to push back if it is made prematurely.
+* For RFCs with lengthy discussion, the motion to FCP is usually preceded by a *summary comment* trying to lay out the current state of the discussion and major tradeoffs/points of disagreement.
+* The FCP lasts ten calendar days, so that it is open for at least 5 business days. It is also advertised widely (i.e. in [Grin News](https://grinnews.substack.com)). This way all stakeholders have a chance to lodge any final objections before a decision is reached.
+* In most cases, the FCP period is quiet, and the RFC is either merged or closed. However, sometimes substantial new arguments or ideas are raised, the FCP is canceled, and the RFC goes back into draft mode.
 
-Eventually, the relevant community organization will either accept the RFC by changing its status to 'Accepted' or reject it by setting it to 'Rejected'.
+#### Active
```

Comment:
I noticed that Rust is light touch here and I think it's intentionally: Once an RFC is merged and "active", a tracking issue is opened in the relevant repo, and that's where all the implementation details and subsequent attention is directed. Which makes sense to me. The RFC itself is therefore is not a full spec and something that should be tracked. Instead becomes something of a historical artefact. 

They don't even have a state for Active/Inactive in Rust, if RFC1 is replaced by RFC2, they simply put a comment in RFC1 that it's been superseded by RFC2 and link to it.

To me, this makes sense, as you ultimately track the actual issues you have in your sub-team's repo.

Reply:
Think I can handle that. I'd say in most cases as development occurs there would be technical issues popping up that might require changing, say an API definition in the original RFC. I assume these kinds of definitions would be worth updating in the RFC as implementation progresses?

Reply:
So this is a really good question. Here's my thinking based on my reading of the Rust process and how I think it makes sense for us to start: 

Let's say there's an API definition (affecting `grin-wallet`) in an RFC that's accepted. A tracking issue for this is created in `grin-wallet`, and then at some point there's work starting to implement this, by opening PRs in `grin-wallet`. And then at some point throughout the course of implementation, some details of the API change.

_Would it be worth while to update the RFC?_ Maybe. Depends on the change and the developer's inclination to do it.

_Should it matter whether the RFC is updated or not?_ By design, **no**. The RFC does not act as a "final spec" or ultimate source of truth. That's what documentation in `grin-wallet` or the code itself is there for. Instead, in the RFC there will be a link to the tracking issue. And in the tracking issue, there will be a link to a PR. And in that PR there will be documentation references (hopefully). And if there's future iterations, one would hope that either the documentation is updated or that future PRs reference that PR. 

The point here is that we should not introduce a process where we expect developers or contributors to keep an RFC document up to date and go back to refer to that as some kind of ultimate reference. As we're going to struggle to maintain that.

Makes sense?

Reply:
Unless the RFC is more of an open specification, where you might have some protocol spec with a link to a maintained reference implementation. But I think it would be fairly evident in that case that the RFC needs to remain updated.